### ISSUE TITLE
Incorrect query frontend warning log when not using memcached cache

### ISSUE THREAD

**Issue by yeya24** [thanos-io/thanos#3240]
**Thanos, Prometheus and Golang version used**:

Thanos master version

**What happened**:

I started the Thanos query frontend with the in-memory response cache. However, there is a warning log about Memcached even if I am not using it.

```
level=info ts=2020-09-27T19:27:28.128182256Z caller=factory.go:38 msg="loading tracing configuration"
level=info ts=2020-09-27T19:27:28.128357861Z caller=jaeger.go:44 msg="loading Jaeger tracing configuration from YAML"
level=info ts=2020-09-27T19:27:28.128900062Z caller=logger.go:18 msg="debug logging disabled"
level=info ts=2020-09-27T19:27:28.129025495Z caller=logger.go:18 msg="debug logging disabled"
level=warn ts=2020-09-27T19:27:28.129805657Z caller=query_frontend.go:110 msg="memcached cache valid time set to 0, so using a default of 24 hours expiration time"
```

**What you expected to happen**:

The log shouldn't show up when using other caches.

**How to reproduce it (as minimally and precisely as possible)**:

```
thanos query-frontend --query-frontend.downstream-url="http://localhost:9091" --query-range.response-cache-config-file=/home/yeya24/res-conf
```

The content of the file `res-conf`

```
type: in-memory
config:
  max_size: "" 
  max_size_items: 2048
  validity: 6h
```

**Anything else we need to know**:

The fix should be pretty straightforward. Just need to move https://github.com/thanos-io/thanos/blob/master/cmd/thanos/query_frontend.go#L109-L112 to https://github.com/thanos-io/thanos/blob/master/pkg/queryfrontend/config.go#L90.

Help wanted!

**Comment by devil-cyber** [comment]
Can i work on this

**Comment by devil-cyber** [comment]
Can i work on this

**Comment by yeya24** [comment]
Sure, please go ahead!

**Comment by devil-cyber** [comment]
Thanks 😊

**Comment by yeya24** [comment]
This is fixed in #3276
